### PR TITLE
Implement region condition data export

### PR DESCRIPTION
**Pull request by david-venhoff**
### Short description

Adds a new admin page with a button to download a region condition export.
The page itself is very minimal and will be improved with #3012.

Speed is not great, unfortunately. An export using production data takes about 6 minutes.

### Proposed changes

- Add a new view
- Add a new endpoint to create a data export
- Move some methods from the dashboard to the region, so that they can be shared by the dashboard and the region condition export

### Side effects

None

### Resolved issues

Fixes: #3053

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/views/region_condition/region_condition_actions.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        model = Region
+        # if we don't define the empty fields all fields are created in the default way additionally to our custom way
+        fields = ()
+
+
+@require_POST
+@permission_required("cms.view_region")
+def export_region_conditions(request: HttpRequest, file_format: str) -> HttpResponse:
+    """
+    Creates a data export summarizing the condition of all region
```
Comment (JoeyStk) [inline integreat_cms/cms/views/region_condition/region_condition_actions.py:150]:
```suggestion
    Creates a data export summarizing the condition of all regions
```

**Comment by MizukiTemma** [comment]
@david-venhoff Thank you for the PR, looks very good 😸 

There is one question: shouldn't the feature only be available for Service Team (+ CMS)? it doesn't harm that App Team and Marketing Team can use this functionality, though it was so specified in the issue 🤔

**Comment by PeterNerlich** [comment]
Had a quick look and for the stock dev test data it takes 43 queries (6 regions). I am confident that this can be significantly reduced, best case it could be one single humongous db query that we just dump as is into the desired format. I think a reduction to one quarter of the time might be realistic.

File: integreat_cms/cms/views/region_condition/region_condition_actions.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+logger = logging.getLogger(__name__)
+
+
+class RegionConditionResource(resources.ModelResource):
+    """
+    A resources class that connects to the django-import-export library.
+    It represents the to-be exported status of all regions.
+    """
+
+    name = fields.Field(column_name=_("Region name"), attribute="name")
```
Comment (PeterNerlich) [inline integreat_cms/cms/views/region_condition/region_condition_actions.py:35]: What do you think, does it make more sense to just say "Region" in this column?
```suggestion
    name = fields.Field(column_name=_("Region"), attribute="name")
```

File: integreat_cms/cms/views/region_condition/region_condition_actions.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    num_pages_with_missing_or_outdated_translation = fields.Field(
+        column_name=_(
+            "Number of pages with at least one missing or outdated translation"
+        )
+    )
+
+    num_outdated_pages = fields.Field(column_name=_("Number of outdated pages"))
+
+    has_translation_package_been_booked = fields.Field(
+        column_name=_("Has translation package been booked"),
```
Comment (PeterNerlich) [inline integreat_cms/cms/views/region_condition/region_condition_actions.py:50]: I think adding a question mark makes it slightly better at conveying the meaning
```suggestion
    has_translation_package_been_booked = fields.Field(
        column_name=_("Has translation package been booked?"),
```

**Review by PeterNerlich — APPROVED** [review]
Well, this was a disappointing investigation of mine :grin: Take my approve and a few over-perfectionist nitpicks and ideas I'm not even sure of myself, since I cannot contribute anything more helpful :stuck_out_tongue:

**Comment by david-venhoff** [comment]
@PeterNerlich Thanks for the review :)

>  I think a reduction to one quarter of the time might be realistic.

I share this sentiment, and I went on to exactly the same investigation as you :smiling_face_with_tear: 
I think that by far the largest bottleneck is in mapping `Link`s to `Region`s and I ~~wasted~~ spent quite a bit of time trying to improve its performance, without much success. For example, it is possible to completely get rid of the `filter_urls` function and perform the calculations completely in the database. However my experiment created such an awful query that it crashed postgres ^^
My current hope is that we can create some indexes in the right places to speed this up, and I created #3077 for it, by I did not start investigating this yet.

**Review by JoeyStk — CHANGES_REQUESTED** [review]
Thank you very much! This looks quite good already. However I noticed a few things when reviewing this with our test data.

**Number of missing or outdated translations**: In our test data I have eight pages that have a missing or outdated translation, the CSV says seven. I have the same error in Augsburg where I had 25 outdated translations for not empty pages, the CSV only says 19.  Maybe it's worth to have a look at [this branch](https://github.com/digitalfabrik/integreat-cms/tree/feature/add-translation-coverage-row-to-dashboard), where attempted at implementing something similar :) 

The rest seemed to be accurate. I think however that thorough testing after merging would be a great benefit :)

File: integreat_cms/cms/views/region_condition/region_condition_actions.py (hunk 2 above)
Comment (david-venhoff) [reply integreat_cms/cms/views/region_condition/region_condition_actions.py:35]: :+1:

**Comment by david-venhoff** [comment]
@MizukiTemma 

> There is one question: shouldn't the feature only be available for Service Team (+ CMS)?

I agree, and it is implemented now

**Comment by JoeyStk** [comment]
> Could you prune the database and try again? In my csv the number is 8 (I assume you refer to nürnberg)

Ok, this is wild. At one point I also had the correct number of 8, but after executing `prune_database.sh` it's now back to 7, but I count 8 missing translations and 5 pages with at least one missing translation (for Nürnberg). I'm not sure what's going on here... For Augsburg the numbers are similarly off (see other screenshot) :thinking: 

Screenshot from 2024-10-08 11-28-14

Screenshot from 2024-10-08 11-20-29

Screenshot from 2024-10-08 11-25-27

**Review by JoeyStk — APPROVED** [review]
Sorry, I was just dumb

**Pull request merged**